### ISSUE TITLE
AddContentProtection functions don't set AdaptationSet field in created object

### ISSUE THREAD

**Issue by juliancooper** [zencoder/go-dash#30]

**Comment by kaihendry** [comment]
I think I've hit this issue whereby ContentProtection tags get stripped.

https://static.spuul.com/devops/2017-11-08/content-protection-stripped.txt


We are Brightcove customers, so we would appreciate some prioritisation here to get this addressed.

**Comment by thomshutt** [comment]
Thanks for the example @kaihendry, I'll take a look today

**Comment by kaihendry** [comment]
In retrospect, I noticed https://github.com/zencoder/go-dash/blob/master/mpd/mpd.go#L90 where you omit empty ContentProtection lines. Tbh perhaps these lines have no value, though I don't understand the mpd/dash very well. Esp when it comes to DRM.

**Comment by thomshutt** [comment]
Nope, this is definitely a bug - it seems that we always lose the ContentProtection block on read.

I'll try to get a PR in with a fix by the end of the day.

**Comment by thomshutt** [comment]
@kaihendry I've got a fix lined up, but there's an underlying Go bug https://github.com/golang/go/issues/9519 with Unmarshalling namespaced XML attributes that's stopping it from working correctly. It's allegedly going to be fixed in the next version of Go, which should be released towards the end of 2017.

**Comment by kaihendry** [comment]
There is no workaround? Thought you get quite firm control if you override the unmarshalling functions. :sweat_smile:
